fix(SidePanel): should fix focus trap which is breaking the side panel

matthewgallo opened the pull request:
Contributes to #434 

This will _hopefully_ fix the side panel focus trap bug that @AlexanderMelox found. Thanks @joshblack for the help 🙏 

#### What did you change?
Line [63-66 from wrapFocus](https://github.com/carbon-design-system/ibm-cloud-cognitive/blob/master/packages/cloud-cognitive/src/global/js/utils/wrapFocus.js#L63-L66) is trying to convert an array like node list to an array using `Array.slice.call`, but it should have been `Array.prototype.slice.call`, which can also be achieved by using the static method `Array.from`.

#### How did you test and verify your work?
- Everything still functions as expected in the storybook for the side panel, however this bug has only been able to be reproduced when using the packaged version of the component.
- New tests for `wrapFocus.js` pass, `jest /utils/`

dcwarwick reviewed (changes requested):
Good catch, and thanks @joshblack for helping track this down.

I notice that we've reworked this logic from the carbon original, presumably to avoid a dependency on lodash for one function alone. That's fine, but I also notice that carbon have a set of tests for wrapFocus.js, and when I run those on the code before this change, the tests fail because of the Array.slice mistake. This change fixes them(*). I feel there's a moral in this story somewhere :-) So I suggest we copy carbon's tests over too -- the file wrapFocus-test.js can just be dropped in to our own /packages/cloud-cognitive/src/global/js/utils/\_\_tests\_\_ folder and will start doing its job right away.

(*) when I say it fixes the tests, it fixes all but one. carbon have added the following lines into wrapFocus.js, with a test case to go along with them. I suggest we also add these lines into our wrapFocus, even if we don't need this function, just so we continue to align with the wrapFocus.js in carbon for easier future comparisons. If these lines are added to our version, all the tests then pass.

![image](https://user-images.githubusercontent.com/6385315/112289357-276ea100-8c86-11eb-9508-6dca3e677112.png)

matthewgallo commented:
Thanks @dcwarwick for the feedback. I added back the last argument for the floating menus and added the tests for this utility from carbon and made sure that everything passes (`jest /utils/`). I definitely agree we should have had these tests for this helper 😅

dcwarwick reviewed (approved):
Looks good - many thanks
